### PR TITLE
Packaging: Try to make New-Unix package more readable

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.
- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Update the changelog](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---changelog)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [NA] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [NA] [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

## PR Summary

Move large chunks of code with-in a single function out into functions to make the function easier to understand.

### CODE REVIEW

**Comment by TravisEz13** [comment]
restarted appveyor due to web timeout

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -586,28 +586,8 @@ function New-UnixPackage {
...
-                    else {
-                        $env:PATH = $originalPath
-                    }
-                }
-
-                throw "Dependency precheck failed!"
-            }
-        }
+        # Verify depenecies are installed and in the path
+        Test-Dependecies
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Typo in `dependencies`, also in the comment above.

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -676,145 +639,54 @@ function New-UnixPackage {
...
-            # Run ronn on assets file
-            # Run does not play well with files named powershell6.0.1, so we generate and then rename
-            Start-NativeExecution { ronn --roff $RonnFile }
-
-            # gzip in assets directory
-            $GzipFile = "$RoffFile.gz"
-            Start-NativeExecution { gzip -f $RoffFile }
-
-            $ManFile = Join-Path "/usr/local/share/man/man1" (Split-Path -Leaf $GzipFile)
+            # Generate gzip of mann file
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: mann -> man

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -886,6 +742,324 @@ function New-UnixPackage {
...
+        [String]$Version,
+
+        [Parameter(Mandatory)]
+        [String]$Iteration,
+
+        [Parameter(Mandatory,HelpMessage='Package description')]
+        [String]$Description,
+
+        [Parameter(Mandatory,HelpMessage='Installer Type')]
+        [String]$Type,
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Please add a comment for the possibilities.

File: tools/packaging/packaging.psm1 [hunk 4]
```diff
@@ -886,6 +742,324 @@ function New-UnixPackage {
...
+        [Parameter(Mandatory,HelpMessage='Staging folder for installation files')]
+        [String]$Staging,
+
+        [Parameter(Mandatory,HelpMessage='Install path on target machine')]
+        [String]$Destination,
+
+        [Parameter(Mandatory,HelpMessage='The built and gzipped mann file.')]
+        [String]$MannGzipFile,
+
+        [Parameter(Mandatory,HelpMessage='The destination of the mann file')]
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Mann -> Man. Also line  770

File: tools/packaging/packaging.psm1 [hunk 5]
```diff
@@ -886,6 +742,324 @@ function New-UnixPackage {
...
+            if (!$Environment.IsMacOS -and !$_)
+            {
+                throw "Must not be null on this environment."
+            }
+            return $true
+        })]
+        [String]$AfterRemoveScript
+    )
+
+    $Arguments = @(
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Consider ArrayList

File: tools/packaging/packaging.psm1 [hunk 6]
```diff
@@ -886,6 +742,324 @@ function New-UnixPackage {
...
+function Test-Distribution
+{
+    param(
+        [String]
+        $Distribution
+    )
+
+    if ( ($Environment.IsUbuntu -or $Environment.IsDebian) -and !$Distribution )
+    {
+        throw "$Distribution is require for a Debian based distribution."
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: require -> required

**Review by adityapatwardhan — APPROVED** [review]
LGTM with minor comments.

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: resolved

File: tools/packaging/packaging.psm1 (hunk 2 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: resolved

File: tools/packaging/packaging.psm1 (hunk 3 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: resolved

File: tools/packaging/packaging.psm1 (hunk 4 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: resolved

File: tools/packaging/packaging.psm1 (hunk 5 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: resolved

Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: This became unreadable after changing to an array list.  I filed this issue to add a feature to PowerShell so I can maintain a similar syntax and use a list:
https://github.com/PowerShell/PowerShell/issues/5643

File: tools/packaging/packaging.psm1 [hunk 7]
```diff
@@ -586,28 +586,8 @@ function New-UnixPackage {
...
-                    }
-                    else {
-                        $env:PATH = $originalPath
-                    }
-                }
-
-                throw "Dependency precheck failed!"
-            }
-        }
+        # Verify depenecies are installed and in the path
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Typo in comment.

File: tools/packaging/packaging.psm1 [hunk 8]
```diff
@@ -850,33 +705,41 @@ function New-UnixPackage {
                     Move-Item $hack_dest $symlink_dest
                 }
             }
-            if ($AfterInstallScript) {
-                Remove-Item -erroraction 'silentlycontinue' $AfterInstallScript
+            if ($AfterScriptInfo.AfterInstallScript) {
+                Remove-Item -erroraction 'silentlycontinue' $AfterScriptInfo.AfterInstallScript
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: -Force maybe?

File: tools/packaging/packaging.psm1 [hunk 9]
```diff
@@ -850,33 +705,41 @@ function New-UnixPackage {
...
             }
-            if ($AfterRemoveScript) {
-                Remove-Item -erroraction 'silentlycontinue' $AfterRemoveScript
+            if ($AfterScriptInfo.AfterRemoveScript) {
+                Remove-Item -erroraction 'silentlycontinue' $AfterScriptInfo.AfterRemoveScript
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: -Force maybe?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/packaging/packaging.psm1 (hunk 9 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: There will be a follow-up PR.  I can address these issue in that PR.

**Comment by adityapatwardhan** [comment]
According to @TravisEz13 the remaining changes will be done in the next PR.

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**